### ISSUE TITLE
Improved mobile navigation layout

### ISSUE THREAD

**Issue by curiousduck-dev** [WordPress/gutenberg#58036]
## What problem does this address?

The navigation block is probably one of the fastest evolving blocks in Gutenberg and it's certainly come a long way. However, in its current form each navigation is siloed, with no ability to interconnect different navigations. For example, each navigation block comes with settings for mobile navigation - whether it be disabled entirely, activated at breakpoint, or always active. If - as is often the case for a great many websites - a header template part contains multiple navigation blocks the current setup results in an ugly mobile state of two separate hamburger icons with two separate mobile navigations.

## What is your proposed solution?
It would be brilliant if some interconnectivity could be established between navigation blocks. In this example, having the ability to add the top navigation block's link list to the main navigation block's mobile navigation would fix the above issue, along with an additional "hide on mobile" setting for the mobile navigation options. That way, on mobile sizes, users could hide the top navigation altogether (removing that ugly second hamburger) and display its link list as part of the main navigation's mobile menu, creating a cleaner UX. 

Here's a screenshot of how things work current on desktop and then on mobile:
<img width="1184" alt="Screenshot 2024-01-20 at 09 42 45" src="https://github.com/WordPress/gutenberg/assets/142537213/a89238c3-e409-4cee-887c-13be4dee95d1">
<img width="393" alt="Screenshot 2024-01-20 at 09 42 58" src="https://github.com/WordPress/gutenberg/assets/142537213/445a727f-5eff-4251-8d30-8b29d9f1ffcc">

**Comment by paaljoachim** [comment]
I will add in this States issue which brings up among other things responsiveness.
https://github.com/WordPress/gutenberg/issues/57719

**Comment by eric-michel** [comment]
I posted a [similar issue](https://github.com/WordPress/gutenberg/issues/56090) last year regarding similar issues with the Navigation block. It feels extremely restrictive in its current form, and navigation is one of the most tailored pieces of many websites.

The Navigation block desperately needs ways to significantly change its output. I have achieves a very nice mobile flyout menu outside of the Navigation block, but it required a lot of workarounds.

**Comment by getdave** [comment]
I agree this situation is sub-optimal. I created [this Plugin](https://wordpress.org/plugins/getdave-responsive-navigation-block/) to provide a workaround.

However, we could dispense with it entirely if something like block visibility was added to Core.

**Comment by nekohayo** [comment]
> However, we could dispense with it entirely if something like block visibility was added to Core.

The irony is that I used the "Block Visibility" plugin to work around the problem of needing differently customized navbar menus for mobile vs desktop, only later finding this ticket here... and _then_ later still discovering that the "Block Visibility" plugin is also [incompatible](https://github.com/ndiego/block-visibility/issues/104#issuecomment-2705424167) with the very popular SuperCache plugin.

It would seem that Automattic would need to coordinate solutions across all three plugins: Gutenberg, Block Visibility, and SuperCache.
